src/common/components/ReviewItem/ReviewItem.tsx

```diff
@@ -0,0 +1,125 @@
+import { Button } from "@/common/components/Button";
+import { reviewItemTheme, type ReviewItemVariants } from "./ReviewItem.theme";
+import { ThumbUpFilledIcon } from "@/common/components/CustomIcon";
+import { LockCtaOverlay } from "@/common/components/LockCtaOverlay";
+import { Tag } from "@/common/components/Tag";
+import { getHumanReadableTimestampDelta } from "@/common/functions";
+
+// TODO: to replace with prisma generated types
+export type Label = {
+  name: string;
+  typeof: "professor" | "course";
+};
+
+export type Review = {
+  body: string;
+  courseCode: string;
+  username: string;
+  likeCount: number;
+  labels: Label[];
+  createdAt: number;
+};
+
+export type ReviewItemProps = ReviewItemVariants & {
+  review: Review;
+  isLocked?: boolean;
+  variant?: "home" | "subpage";
+};
+
+export const ReviewItem = ({
+  review,
+  isLocked,
+  variant = "home",
+}: ReviewItemProps) => {
+  const {
+    wrapper,
+    headingContainer,
+    schoolContainer,
+    schoolIcon,
+    schoolCourseCode,
+    metadataContainer,
+    profileContainer,
+    profileName,
+    profileIcon,
+    likeIcon,
+    timedelta,
+    body,
+    labels,
+  } = reviewItemTheme({ variant });
+
+  const HeaderCourse = () => (
+    <div className={schoolContainer()}>
+      <div className={schoolIcon()}>
+        <div className="h-4 w-4 rounded-full bg-red-800"></div>
+      </div>
+      <div className={schoolCourseCode()}>{review.courseCode}</div>
+    </div>
+  );
+
+  const HeaderReviewer = () => (
+    <div className={profileContainer()}>
+      <div className={profileIcon()}>
+        <div className="h-4 w-4 rounded-full bg-cyan-800"></div>
+      </div>
+      <div className={profileName()}>{review.username}</div>
+    </div>
+  );
+
+  const ReviewContent = ({ isDetailed }: { isDetailed: boolean }) => {

```

Comment:
I think we can move this as its own component under `ReviewItem/ReviewContent`

Reply:
It is generally not advisable to declare components within a component i.e. `ReviewContent` inside of `ReviewItem` due it re-rendering whenever ReviewItem changes.

Maybe all the components like `HeaderReviewer` can be renamed and moved to a subfolder like `ReviewItem/ReviewerProfile`. Wdyt?

Reply:
updated at 2eaf9f01!